programs/whirlpool/src/math/swap_math.rs

```diff
@@ -63,15 +65,17 @@ pub fn compute_swap(
     )?;
 
     // If the swap is not at the max, we need to readjust the amount of the fixed token we are using
-    if !is_max_swap {
-        amount_fixed_delta = get_amount_fixed_delta(
+    let amount_fixed_delta = if !is_max_swap || initial_amount_fixed_delta.exceeds_max() {
```

Comment:
nit: can we add a comment to explain that we are capping the amount delta here?

Reply:
yep.
I added detailed comment on the head of the function, and added comment here to clarify that the result (`amount_fixed_delta`) will be in the u64 range.